core/src/main/java/org/apache/accumulo/core/file/rfile/RFileOperations.java

```diff
@@ -82,7 +82,13 @@ protected FileSKVWriter openWriter(OpenWriterOperation options) throws IOExcepti
     AccumuloConfiguration acuconf = options.getTableConfiguration();
 
     long blockSize = acuconf.getAsBytes(Property.TABLE_FILE_COMPRESSED_BLOCK_SIZE);
+    if (blockSize >= Integer.MAX_VALUE || blockSize < 0) {
+      throw new IllegalArgumentException("table.file.compress.blocksize must be greater than 0 and less than " + Integer.MAX_VALUE);
```

Comment:
I think there might be a Guava `Preconditions.checkArgument(boolean check, String message)` which would simplify these IllegalArgumentException checks and throws.